snuba/datasets/errors_replacer.py

```diff
@@ -537,38 +535,106 @@ def process_merge(
     )
 
 
+def _convert_hash_list(
+    hashes: Sequence[str], state_name: ReplacerState, convert_types: bool = False
+) -> str:
+    if state_name == ReplacerState.ERRORS:
+        if convert_types:
+            return ", ".join(
+                "toUUID('%s')" % str(uuid.UUID(_hashify(h))) for h in hashes
+            )
+        else:
+            return ", ".join("'%s'" % str(uuid.UUID(_hashify(h))) for h in hashes)
+    else:
+        if convert_types:
+            return ", ".join("toFixedString('%s', 32)" % _hashify(h) for h in hashes)
+        else:
+            return ", ".join("'%s'" % _hashify(h) for h in hashes)
+
+
+def _build_grouping_hashes_filter(
+    message: Mapping[str, Any], state_name: ReplacerState
+) -> Optional[Tuple[List[str], List[str], MutableMapping[str, str]]]:
+    hashes = message.get("hashes") or []
+    hierarchical_hashes = message.get("hierarchical_hashes") or {}
```

Comment:
Can you elaborate more on what this dictionary is? What are the keys and what are the values?

Reply:
hierarchical_hashes is a mapping from an entry in `hierarchical_hashes` snuba column to the corresponding primary_hash.

there's a code comment further down that should explain what this is, lmk if I should move it elsewhere or clarify it

Reply:
The comment doesn't explain what this data structure is, so that might be helpful. I was struggling to figure out what all the different hashes are supposed to be.